code/modules/mining/equipment/wormhole_jaunter.dm

```diff
@@ -89,3 +89,92 @@
+	icon_state = "flare-contractor"
+	item_state = "flare"
+	var/destination
+
+/obj/item/wormhole_jaunter/contractor/verb/set_destination()
+	set category = "Object"
+	set name = "Change Portal Destination"
+	set src in usr
+
+	var/list/L = list()
+	var/list/areaindex = list()
+
+	for(var/obj/item/radio/beacon/R in GLOB.beacons)
+		var/turf/T = get_turf(R)
+		if(!T)
+			continue
+		if(!is_teleport_allowed(T.z))
+			continue
+		var/tmpname = T.loc.name
+		if(areaindex[tmpname])
+			tmpname = "[tmpname] ([++areaindex[tmpname]])"
+		else
+			areaindex[tmpname] = 1
+		L[tmpname] = R
+
+	var/desc = input("Please select a location to target.", "Flare Target Interface") in L
+	destination = L[desc]
+	return
+
+/obj/item/wormhole_jaunter/contractor/attack_self(mob/user) // message is later down
+	activate(user, TRUE)
+
+/obj/item/wormhole_jaunter/contractor/activate(mob/user)
+	if(!turf_check(user))
+		return
+	if(!destination)
+		var/list/L = get_destinations(user)
+		if(!L.len)
+			to_chat(user, "<span class='notice'>[src] found no beacons in the sector to target.</span>")
+			return
+		destination = pick(L)
+	var/obj/effect/temp_visual/getaway_flare/F = new(get_turf(src))
+	user.visible_message("<span class='notice'>[user] pulls out a black and gold flare and lights it.</span>",\
+						  "<span class='notice'>You light an emergency extraction flare, initiating the extraction process.</span>")
+	user.drop_item()
+	forceMove(F)
+	addtimer(CALLBACK(src, .proc/create_portal, destination), 5 SECONDS)
+
+/obj/item/wormhole_jaunter/contractor/proc/create_portal(turf/destination)
+	new /obj/effect/portal/redspace/getaway(get_turf(src), get_turf(destination), src, 100)
+	qdel(src)
+
+/obj/item/wormhole_jaunter/contractor/emag_act(mob/user)
+	to_chat(user, "<span class='warning'>Emagging [src] has no effect.</span>")
+
+/obj/item/wormhole_jaunter/contractor/chasm_react(mob/user)
+	return //This is not an instant getaway portal like the jaunter
+
+/obj/item/storage/box/syndie_kit/escape_flare
+	name = "emergency extraction kit"
+
+/obj/item/storage/box/syndie_kit/escape_flare/New()
+	..()
+	new /obj/item/wormhole_jaunter/contractor(src)
+	new /obj/item/radio/beacon/emagged(src)
+	return
+
+/obj/effect/portal/redspace/getaway
+	one_use = TRUE
+
+/obj/effect/temp_visual/getaway_flare // Because the original contractor flare is not a temp visual, for some reason.
+	name = "contractor extraction flare"
+	icon = 'icons/obj/lighting.dmi'
+	icon_state = "flare-contractor-on"
+	duration = 51 // Needs to be slightly longer then the callback to make the portal
+
+/obj/effect/temp_visual/getaway_flare/New()
+	..()
+	playsound(loc, 'sound/goonstation/misc/matchstick_light.ogg', 50, TRUE)
+	set_light(8, l_color = "#FFD165")
+
+/obj/effect/temp_visual/getaway_flare/Destroy()
+	new /obj/effect/decal/cleanable/ash(loc)
+	return ..()
```

Comment:
This proc is really meant for the GC cleaning things up, not for when an object disappears (IE: This is called whenever something is deleted, even by admins).


I would much rather this be part of the duration thing, though I dont know of any better way to do this on the top of my head.

Reply:
Moved to the callback that makes the portal, removes the destroy check